packages/nimble-components/src/patterns/button/styles.ts

```diff
@@ -257,84 +265,156 @@ export const buttonAppearanceVariantStyles = css``.withBehaviors(
         ButtonAppearance.outline,
         css`
             :host([appearance-variant='primary']) .control {
-                box-shadow: 0px 0px 0px ${borderWidth}
-                    rgba(${actionRgbPartialColor}, 0.3) inset;
+                border-color: ${actionRgbPartialColor};
             }
 
             :host([appearance-variant='primary']) .control:hover {
-                box-shadow: 0px 0px 0px ${borderWidth} ${borderHoverColor} inset;
+                border-color: ${borderHoverColor};
+                background-color: transparent;
             }
 
             :host([appearance-variant='primary']) .control${focusVisible} {
-                box-shadow: 0px 0px 0px ${borderWidth} ${borderHoverColor} inset;
+                border-color: ${borderHoverColor};
             }
 
             :host([appearance-variant='primary']) .control:active {
-                box-shadow: 0px 0px 0px ${borderWidth} ${fillSelectedColor}
-                    inset;
+                outline: none;
             }
 
             :host([appearance-variant='primary'][disabled]) .control {
-                box-shadow: none;
+                border-color: rgba(${borderRgbPartialColor}, 0.3);
+            }
+
+            :host([appearance-variant='accent']) .control {
+                border-color: ${buttonBorderAccentOutlineColor};
+                color: ${buttonAccentOutlineFontColor};
+            }
+
+            :host([appearance-variant='accent']) .control:hover {
+                background-color: transparent;
+            }
+
+            :host([appearance-variant='accent']) .control:active {
+                outline: none;
+            }
+
+            :host([appearance-variant='accent'][disabled]) .control {
+                border-color: rgba(${borderRgbPartialColor}, 0.3);
+                color: rgba(${actionRgbPartialColor}, 0.3);
             }
         `
     ),
     appearanceBehavior(
         ButtonAppearance.block,
         css`
-            :host([appearance-variant='primary']) .control {
-                border-color: rgba(${actionRgbPartialColor}, 0.3);
-                box-shadow: 0px 0px 0px ${borderWidth}
-                    rgba(${actionRgbPartialColor}, 0.3) inset;
+            :host([appearance-variant='primary']) [part='start'] {
+                ${iconColor.cssCustomProperty}: white;
```

Comment:
Right now we duplicate `${iconColor.cssCustomProperty}: white;` across many rules. Should we consider ways to reduce that duplication or is it needed because the value changes so much depending on the configuration of the button?

Ideas to consider:
- some theme aware tokens for icon color and disabled icon color
- local CSS variables for the same
- finding ways to apply the color on a less specific selector. If it only changes based on theme and disabled state then maybe we don't need to set it in rules that target specific appearance variants

Reply:
My most recent commit has a solution that Mert came up with to fix this, which was cascade layers for the styles of the button. By doing this, a hierarchy could be created that allowed the `${iconColor.cssCustomProperty}: white;` along with other repeated styles to be removed since they don't have to be reiterated with cascade layers.

These layers have to be wrapped around every css class which could be unwanted, but at the same time it helps get rid of repeating css that could be confusing. The hierarchy of the layers follows the [css guidelines](https://github.com/ni/nimble/blob/main/packages/nimble-components/docs/css-guidelines.md#prefer-cascade-for-state-changes).